### PR TITLE
Add retry to exec measurement

### PR DESCRIPTION
**Pull request by aleksandra-malinowska**
Add support for configuring number of retries and interval between retries to exec measurement.

/kind feature

### CODE REVIEW

File: clusterloader2/pkg/measurement/common/exec.go [hunk 1]
```diff
@@ -62,22 +62,40 @@ func (e *execMeasurement) Execute(config *measurement.Config) ([]measurement.Sum
...
+		return nil, fmt.Errorf("error getting retries, retries: %v, err: %v", retries, err)
+	}
+	retryIntervalStr, err := util.GetStringOrDefault(config.Params, "retryInterval", "1s")
+	if err != nil {
+		return nil, err
+	}
+	retryInterval, err := time.ParseDuration(retryIntervalStr)
+	if err != nil {
+		return nil, err
+	}
```
Comment (tosi3k) [inline clusterloader2/pkg/measurement/common/exec.go:76]: `GetDurationOrDefault` will simplify this to a single error check.

File: clusterloader2/pkg/measurement/common/exec.go [hunk 2]
```diff
@@ -62,22 +62,40 @@ func (e *execMeasurement) Execute(config *measurement.Config) ([]measurement.Sum
...
+		klog.V(2).Infof("Running %v with timeout %v, attempt %v", command, timeout, i+1)
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
+		cmd := exec.CommandContext(ctx, command[0], command[1:]...)
+		out, err := cmd.CombinedOutput()
+		klog.V(2).Infof("Exec command output: %v", string(out))
+		if err == nil {
+			klog.V(2).Infof("Command %v succeeded in attempt %v", command, i+1)
+			return nil, nil
+		}
```
Comment (tosi3k) [inline clusterloader2/pkg/measurement/common/exec.go:93]: I'd probably still print out the errors if the past retries failed.

**Review by tosi3k — CHANGES_REQUESTED** [review]

File: clusterloader2/pkg/measurement/common/exec.go (hunk 2 above)
Comment (aleksandra-malinowska) [reply clusterloader2/pkg/measurement/common/exec.go:85]: Done

File: clusterloader2/pkg/measurement/common/exec.go (hunk 1 above)
Comment (aleksandra-malinowska) [reply clusterloader2/pkg/measurement/common/exec.go]: Done (here and for timeout)

File: clusterloader2/pkg/measurement/common/exec.go [hunk 3]
```diff
@@ -62,22 +58,37 @@ func (e *execMeasurement) Execute(config *measurement.Config) ([]measurement.Sum
 	if len(command) == 0 {
 		return nil, fmt.Errorf("command is a required argument. Got empty slice instead")
 	}
-
+	retries, err := util.GetIntOrDefault(config.Params, "retries", 1)
+	if err != nil || retries < 1 {
+		return nil, fmt.Errorf("error getting retries, retries: %v, err: %v", retries, err)
+	}
+	retryInterval, err := util.GetDurationOrDefault(config.Params, "retryInterval", 1*time.Second)
```
Comment (tosi3k) [inline clusterloader2/pkg/measurement/common/exec.go:65]: Maybe `s/retryInterval/backoffDelay/g`?

Comment (aleksandra-malinowska) [reply clusterloader2/pkg/measurement/common/exec.go]: Done

**Pull request merged**